var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -13,6 +13,15 @@ class Charliecloud(MakefilePackage):
     url      = "https://github.com/hpc/charliecloud/releases/download/v0.9.10/charliecloud-0.9.10.tar.gz"
     git      = "https://github.com/hpc/charliecloud.git"
 
+    depends_on('gcc')
+    depends_on('make')
+    depends_on('rsync')
+    depends_on('python@3')
+    depends_on('py-lark-parser')
+    depends_on('py-sphinx')
+    depends_on('py-sphinx-rtd-theme')
```

Comment:
Any documentation-related dependencies should be optional, and off by default. Very few people actually read installed documentation (especially html) when it is available online, and we want to minimize the number of dependencies where we can.

Reply:
Python3 and the lark parser module are needed for charliecloud's unprivileged builder introduced in v0.10. It is unclear how to require these packages for charliecloud versions 0.10 and beyond.

Python and Sphinx are also used to build our man pages. Excluding it means zero documentation. If these can be offered as optional dependencies, can you advise how to accomplish this? The variant examples seem to be tied to autotools and configure options.

Reply:
What happens when you install the package and these dependencies aren't present? Does the build fail, or is the documentation just skipped? Is there an environment variable or make argument that can be used to explicitly enable/disable the documentation?

To add conditional dependencies, you can do something like:
```python
variant('docs', default=False, description='Build the documentation')

depends_on('py-sphinx', when='+docs')
```

Reply:
The documentation build is skipped in that case.